torch/csrc/jit/runtime/register_cuda_ops.cpp

```diff
@@ -67,6 +67,15 @@ RegisterOperators const reg({
         [](Stack* stack) {
           auto v = pop(stack);
           auto s = v.toCustomClass<torch::jit::CUDAStream>();
+          auto stream_device_idx = static_cast<int64_t>(s->device_index());
+          auto cur_device_idx =
+              static_cast<int64_t>(c10::cuda::current_device());
```

Comment:
Consider using `c10::DeviceIndex` here since we use `int64_t` mainly to make Torchbind happy.

```suggestion
          auto stream_device_idx = static_cast<c10::DeviceIndex>(s->device_index());
          auto cur_device_idx =
              static_cast<c10::DeviceIndex>(c10::cuda::current_device());
```